### PR TITLE
Plot method for aggregation + refactor tests

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Partial #1406 
* Add to aggregation metrics
* Add to binary confusion matrix + multilabel confusion matrix (missing from last pr)
* Refactor test such that `.plot` method is called instead of the underlying function to make sure the complete traceback is tested

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/aggregation.py [hunk 1]
```diff
@@ -153,6 +158,51 @@ def update(self, value: Union[float, Tensor]) -> None:  # type: ignore
...
+        """Plot a single or multiple values from the metric.
+
+        Args:
+            val: Either a single result from calling `metric.forward` or `metric.compute` or a list of these results.
+                If no value is provided, will automatically call `metric.compute` and plot that result.
+            ax: An matplotlib axis object. If provided will add plot to that axis
+
+        Returns:
+            fig: Figure object
+            ax: Axes object
```
Comment (Borda) [inline src/torchmetrics/aggregation.py:173]:
```suggestion
        Returns:
            Figure and Axes object
```

File: src/torchmetrics/aggregation.py [hunk 2]
```diff
@@ -214,6 +264,51 @@ def update(self, value: Union[float, Tensor]) -> None:  # type: ignore
...
```
Comment (Borda) [inline src/torchmetrics/aggregation.py:279]:
```suggestion
            Figure and Axes object
```

File: src/torchmetrics/aggregation.py [hunk 3]
```diff
@@ -273,6 +368,52 @@ def update(self, value: Union[float, Tensor]) -> None:  # type: ignore
...
```
Comment (Borda) [inline src/torchmetrics/aggregation.py:383]:
```suggestion
            Figure and Axes object
```

File: src/torchmetrics/aggregation.py [hunk 4]
```diff
@@ -407,3 +548,48 @@ def update(self, value: Union[float, Tensor], weight: Union[float, Tensor] = 1.0
...
```
Comment (Borda) [inline src/torchmetrics/aggregation.py:564]:
```suggestion
            Figure and Axes object
```

File: src/torchmetrics/classification/confusion_matrix.py [hunk 5]
```diff
@@ -125,6 +129,40 @@ def compute(self) -> Tensor:
...
+
+        Args:
+            val: Either a single result from calling `metric.forward` or `metric.compute` or a list of these results.
+                If no value is provided, will automatically call `metric.compute` and plot that result.
+            add_text: if the value of each cell should be added to the plot
+            labels: a list of strings, if provided will be added to the plot to indicate the different classes
+
+        Returns:
+            fig: Figure object
+            ax: Axes object
```
Comment (Borda) [inline src/torchmetrics/classification/confusion_matrix.py:145]:
```suggestion
            Figure and Axes object
```

File: src/torchmetrics/classification/confusion_matrix.py [hunk 6]
```diff
@@ -351,6 +393,40 @@ def compute(self) -> Tensor:
...
```
Comment (Borda) [inline src/torchmetrics/classification/confusion_matrix.py:409]:
```suggestion
            Figure and Axes object
```

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki could you pls check the conflicts... :)

**Review by stancld — APPROVED** [review]

**Pull request merged**